pkgs/by-name/av/avdump3/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  writeScriptBin,
+  fetchzip,
+  dotnet-runtime,
+  zlib,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "avdump3";
+  version = "8293_stable";
+
+  src = fetchzip {
+    url = "https://cdn.anidb.net/client/avdump3/avdump3_8293_stable.zip";
+    hash = "sha256-H9Sn3I4S9CmymKIMHVagDy+7svHs285S3EJgYQo+ks0=";
+    stripRoot = false;
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/share/avdump3 $out/bin
+    mv * $out/share/avdump3
+    cat > $out/bin/avdump3 <<EOF
+    #!/bin/sh
```

Comment:
nit: i'm a little against this style of shebang, even if it _would_ work.

there's stylistic reasons and there's a logical reason as well. stylistically, i think having a direct path shebang is contrary to nixpkgs' whole goal of declarativeness. logically, while this shebang will have relatively consistent behavior on NixOS, on other distros that use Nix as a package manager `/bin/sh` can point to the literal `sh` executable, an older version of `bash`, something minimal like `dash`, etc etc.

Can you instead change this whole bit into a `writeShellScript` style derivation?

Reply:
I'm not sure what exactly you're suggesting, but would it be acceptable to simply replace `#!/bin/sh` with `#!${runtimeShell}` in this inline script? That seems the simplest, and it's a pattern that occurs elsewhere in nixpkgs.

Reply:
I implemented my above suggestion.